### DISCUSSION TITLE
Some minor feedback

### DISCUSSION THREAD

**Discussion by Hyperlooper** [Ironclad/rivet#284] [General]
As someone who is not super technical in terms of coding ability, I have tested many tools for working with LLM's in a low-code environment.  (langflow, flowise et. al.). 

Rivet is BY FAR the best solution I have found so far. The inclusion of logic & flow control, the superior UI, and the ability to treat subgraphs as reusable functions are the standout features so far. Just wanted to say thank you to the people who have had a hand in developing this, it is already super useful and I am so happy I have found Rivet. 

In my line of work I have interacted with many node-based tools, and assisted with UX for several 'low code' toolchains. Based on the PR's it looks like hotkey support is on the roadmap so here's a few thoughts:

1. Keyboard support for copy/paste & delete would speed up workflows quite a bit.
2. Panning with MMB is quite standard across many, if not most, node-editor tools. Left-Click is a very overloaded input, so using either MMB or  "<modifier key> + LMB" for panning  should probably be considered for default.
3. SHIFT-LMB to select multiple nodes is a very standard UX pattern for node based tools, I find that I miss this quite a lot in Rivet. Marquee tool is useful, but not a replacement for SHIFT-SELECT, in my opinion. 

Anyway, hope this is helpful. If not feel free to ignore :) 

-Trev

**Comment by abrenneke** [comment]
Thank you for the kind words! They mean a lot!

> Keyboard support for copy/paste & delete would speed up workflows quite a bit.

There is copy/paste support with cmd+c,cmd+v. Are you on Windows? Hotkeys have been kind of broken there. The delete key would be a great addition, can you open an issue?

> Panning with MMB is quite standard across many, if not most, node-editor tools. Left-Click is a very overloaded input, so using either MMB or " + LMB" for panning should probably be considered for default.

Yeah you're right. I think at this point it might have to be a setting to disable the left-click-drag. Can you open an issue for the middle-mouse-pan? That would be great!

> SHIFT-LMB to select multiple nodes is a very standard UX pattern for node based tools, I find that I miss this quite a lot in Rivet. Marquee tool is useful, but not a replacement for SHIFT-SELECT, in my opinion.

You totally can do this, you just have to click the title of the node instead of the body. You could open an issue for shift-clicking the body, I think that would be possible.
